src/ansi.rs

```diff
@@ -0,0 +1,1912 @@
+    }
+    let mut num: u8 = 0;
+    for c in input {
+        let c = *c as char;
+        if let Some(digit) = c.to_digit(10) {
+            num = match num.checked_mul(10).and_then(|v| v.checked_add(digit as u8)) {
+                Some(v) => v,
+                None => return None,
+            }
+        } else {
+            return None;
+        }
+    }
+    Some(num)
+}
+
+/// Internal state for VTE processor.
+#[derive(Debug, Default)]
+struct ProcessorState<T: SyncHandler> {
+    /// Last processed character for repetition.
+    preceding_char: Option<char>,
+
+    /// DCS sequence waiting for termination.
+    dcs: Option<Dcs>,
+
+    /// State for synchronized terminal updates.
+    sync_state: SyncState<T>,
+}
+
+#[derive(Debug)]
+struct SyncState<T: SyncHandler> {
+    /// Handler for synchronized updates.
+    handler: T,
+
+    /// Sync DCS waiting for termination sequence.
+    pending_dcs: Option<Dcs>,
+
+    /// Bytes read during the synchronized update.
+    buffer: Vec<u8>,
+}
+
+impl<T: SyncHandler> Default for SyncState<T> {
+    fn default() -> Self {
+        Self {
+            buffer: Vec::with_capacity(SYNC_BUFFER_SIZE),
+            pending_dcs: None,
+            handler: T::default(),
+        }
+    }
+}
+
+/// Pending DCS sequence.
+#[derive(Debug)]
+enum Dcs {
+    /// Begin of the synchronized update.
+    SyncStart,
+
+    /// End of the synchronized update.
+    SyncEnd,
+}
+
+/// The processor wraps a `crate::Parser` to ultimately call methods on a Handler.
+#[derive(Default)]
+pub struct Processor<T: SyncHandler = DefaultSyncHandler> {
+    state: ProcessorState<T>,
+    parser: crate::Parser,
+}
+
+impl<T: SyncHandler> Processor<T> {
+    #[inline]
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Process a new byte from the PTY.
+    #[inline]
+    pub fn advance<H>(&mut self, handler: &mut H, byte: u8)
+    where
+        H: Handler,
+    {
+        if !self.state.sync_state.handler.pending_timeout() {
+            let mut performer = Performer::new(&mut self.state, handler);
+            self.parser.advance(&mut performer, byte);
+        } else {
+            self.advance_sync(handler, byte);
+        }
+    }
+
+    /// End a synchronized update.
+    pub fn stop_sync<H>(&mut self, handler: &mut H)
+    where
+        H: Handler,
+    {
+        // Process all synchronized bytes.
+        for i in 0..self.state.sync_state.buffer.len() {
+            let byte = self.state.sync_state.buffer[i];
+            let mut performer = Performer::new(&mut self.state, handler);
+            self.parser.advance(&mut performer, byte);
+        }
+
+        // Resetting state after processing makes sure we don't interpret buffered sync escapes.
+        self.state.sync_state.buffer.clear();
+        self.state.sync_state.handler.update_timeout(None);
+    }
+
+    /// Number of bytes in the synchronization buffer.
+    #[inline]
+    pub fn sync_bytes_count(&self) -> usize {
```

Comment:
Did you just *remove* the `sync_timeout` method without any replacement?!

Reply:
https://github.com/alacritty/vte/pull/85/commits/f086eb7137b8191ddb4597a75939066968c25334